tests/test_env.py

```diff
@@ -63,6 +63,15 @@ def test_env_accessors(gdalenv):
     with pytest.raises(EnvError):
         getenv()
 
+
+def test_no_aws_gdal_config(gdalenv):
+    """Trying to set AWS-specific GDAL config options fails."""
+    with pytest.raises(EnvError):
+        Env(AWS_ACCESS_KEY_ID='x')
+    with pytest.raises(EnvError):
+        Env(AWS_SECRET_ACCESS_KEY='y')
```

Comment:
@geowurster as you suggested, I'm intercepting attempts to set these options and raising `EnvError`. Let's always use boto3 for credentials. It's in the docstring.